worker_tasks/categorize.py

```diff
@@ -1,11 +1,44 @@
-from collections import Counter
+# categorize.py
+# 
+# Adapted from algorithm by Grineva, et. al. presented at:
+# http://www.slideshare.net/maria.grineva/extracting-key-terms-from-noisy-and-multitheme-documents?related=1
+# 
+# Algorithm Overview
+# 1) Get initial list of possible keyword candidates using ngram frequencies
+# 2) For each keyword candidate, use Wikipedia to extract all related article titles
+#       a) Related articles are called (disambiguation links) on Wikipedia
+# 3) Try to assign one Wikipedia article to each keyword candidate
+#       a) First, assign articles to the keywords with only 1 related article
+#       b) Use these "auto-assigned" articles as the context to narrow the other categories' article
```

Comment:
The context is the set of all "auto-assigned" titles. We can't go outside this set because we need  actual articles to compute relevance, since that score relies on the number of overlapping Wikipedia articles.